### ISSUE TITLE
Deep Linking- MVP (Secure Messages)

### ISSUE THREAD

**Issue by kellylein** [department-of-veterans-affairs/va-mobile-app#6037]
# Project Status Overview

### Project Objective:

**Goal:** Implement Deep Linking Functionality for the Mobile Application.
**_See Project Scope Below for Details._**

[Product Brief](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-app/features/Deep%20Linking/Product%20Brief.md)
 
### Current Dependencies / Blockers
<details>
<summary>Blockers</summary>

|Blocker | Team / Owner | Est to Resolution |
|-------|---------|-------| 
|N/A |   |
|  |   |
|  |   |

</details>

### Project Timeline
<details>
<summary>Milestones</summary>

|Step | Timing | 
|-------| ---------| 
|Sprint 0 - Discovery |Q3-2023   |
|Step 1 - Strategy | Q3-2023  |
|Step 2 - Design  |  Q3-2023 |
|Step 3 - Building / Implementing |Q3-2023   |
| 100% Release |2.10.0 Release- August 29, 2023  |
|Step 4 - Post Verification|   |
</details>

### Monthly Update 
<details>
<summary>Updates</summary>

|Month| Project Health | Progress & Key Accomplishments| Issues/Risks/Blockers|
|-------| ---------|--------|---------|
|July '23| N/A
|Aug '23| Released 2.10.0 
|Sep '23|
|Oct '23|
</details>

# Detailed Project Scope 

### Problem Statement

Today, the Deep Linking functionality does not exist in the Mobile App.
We  would like the mobile user to have the ability to follow a single link (via push notification) into a specific area within the mobile app.

**Product Brief**  [HERE](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-app/features/Deep%20Linking/Product%20Brief.md) (**MVP ONLY**)

**MVP Decisions (High Level) for Q3:**

1. **Secure Messaging vs Appointments**- Deep Linking via **SECURE MESSAGING for MVP** There was a higher volume of users was noted for appointments, however we would like to go with secure messaging as it poses the least complication overall for MBP

2. **Deep Link to Indiv Message in the User Inbox**: User receives a push notification and this will “deep link’ to the individual message in the inbox (latest message at the top)

### Definition of Done 
<details>
<summary>Following needs to be true</summary>

The project will be considered "done" once the following is achieved

1. Items been tested, shipped and released
2. Analytics Dashboards built and in place (monitoring is continuous)

|Item | Completed | 
|-------| ---------|
|August 2023 |Released 2.10.0 | 
| August 2023/September 2023| Analytics Dashboard Set Up in Place| 

</details>

### Project Scope
<details>
<summary>Items in and out of scope</summary>

|In Scope | Out of Scope| 
|-------| ---------|
| | | 
| | | 

In Scope Items for MVP:

MVP Decisions (High Level) for Q3:

1. Secure Messaging vs Appointments- Deep Linking via SECURE MESSAGING for MVP There was a higher volume of users was noted for appointments, however we would like to go with secure messaging as it poses the least complication overall for MBP

2. Deep Link to Indiv Message in the User Inbox: User receives a push notification and this will “deep link’ to the individual message in the inbox (latest message at the top)

</details>

### Important Project Links 
<details>
<summary>Links</summary>

- [Github](url)
- [Product Brief]((https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/va-mobile-app/features/Deep%20Linking/Product%20Brief.md) (**MVP ONLY**))
- [Product Canvas](url)
- Design 
   - [Discovery](url)
   - [Mockups](url)
   - [Final Design](url)
   - [Figma File](url)
- Research
   - [Document 1](url) 
   - [Document 2 ](url) 
-Analytics
   - [Document 1](url) 
</details>

### Team Members

<details>
<summary>Contact List</summary>

- **OCTO Product Leads:** Rachel Han and Ryan Thurlwell
- **PM:** Ryan Thurwell
- **Ad Hoc PM:** Stacy Blackwood
- **Engineering:** Main: Theo Benton , Additional: Jayson Perkins and Jon Bindbeutel
- **Design:** Holly Collier
- **External Team 1 & Role:**
- **External Team 1 & Role:**

</details>

### OKR and Analytic Metrics 
<details>
<summary>Metrics</summary>

|What we are measuring | Why | Event Name 

The following new analytics events and parameters are required to measure the efficacy of the MVP of deeplinking in Secure Messaging:

1.  vama_notification_click: triggered when a user clicks any notification
2.  parameter notification_url the link from the notification

</details>

# Checklist

<details>
  <summary>Toggle Product Checklist</summary>

**Project Kickoff & Requirements Gathering**
   - [X] Project Kickoff session
   - [X ] Initiative Scope finalized
   - [X] Link Product Brief - [Sprint 0 ](https://docs.google.com/document/d/1WHDJ1iCt9J-acLx8_DSL4tkRKoVYqJGaRUqXuFf7ceg/edit# )
   - [ ] Link to Implementation Strategy- N/A
       - Implementation Plan
       - UAT Plan
       - Release Plan
       - Metrics Identified (what metrics are we capturing, how do they tie into stakeholder OKRs, etc)

**Pre-Production Readiness**
- [X] Associate Readiness
- [X] Update product guide-N/A
- [X] Coordinate with contact center support teams-N/A
- [X] What’s New Content- N/A
- [X] App Store/In App messaging (if applicable)- N/A
- [X] Coordinate with Release coordinator
- [X] App store Content changes required? No
- [X] App store images needed? No
- [X] QA Signed off 
- [X] UAT complete
- [X] Obtain Go/No Go decision

**Feature Go Live**
- [X] Ensure Waygate/Feature toggle release ticket submitted w/engineering & QA - N/A
  - **Template: Turn on Waygate/Turn off feature flag**
- [ ] Coordinate with content & release coordinator if what’s new in app is required
</details>

<details>
  <summary>Toggle Team Collaboration Checklist</summary>

-[X ] Design Intent & Feasibility discussions
-[X] Research needed? No
-[X] UAT Needed? No
-[X] Ticket creation -N/A
-[X] Demo feature (mid check, end of development)-N/A
</details>

<details>
  <summary>Toggle Design/UX Checklist</summary> (Not Applicable)

- [ ] Review problem/opportunity statement
- [ ] Add questions and assumptions to product brief
- [ ] Add research links to product brief
- [ ] Document high-level research/design plans
- [ ] Create lo/hi fi wireframes & user flows if applicable
- [ ] Add links to Epic & applicable GitHub folders
- [ ] Socialize designs
- [ ] Sign off received
- [ ] Complete Design QA
- [ ] Does it require component review?
- [ ] Accessibility review needed?

</details>

<details>
  <summary>Toggle Content Checklist</summary> (Not Applicable)

- [ ] Review product brief and surface any questions, assumptions & risks
- [ ] Participate in project discovery and kickoff activities/ceremonies
- [ ] Complete comparative analysis/content research activities for net-new content
- [ ] Review current content and determine where app may need to differ and potential improvements to share with web
- [ ] Review past VA research and decisions documentation
- [ ] Support Research & UX Design in research sessions and synthesis
- [ ] Make content recommendations for lo-/hi-fi wireframes in collaboration with UX Design
- [ ] Collaborate with Sitewide Content team for alignment, improvements, and sign-off
- [ ] Participate in FE hand off
- [ ] Complete content QA
- [ ] Write copy for What's New In App
- [ ] Write copy for app stores' What's New sections.
- [ ] Write copy for updated screenshots in app stores 

</details>

<details>
  <summary>Toggle Engineering Checklist</summary> (Not Applicable)

- [ ] Are UI designs/ specs available/ready? If yes, they should be attached to  attach specs. (We are assuming the design specs have been approved)
- [ ] Are there any deadlines that need to be met?
- [ ] What is the roll out plan? (Do we need a feature toggle? is this going to be available to every user immediately or do we want to roll out to a few users first)
- [ ] Does this project require api integration? if yes, what api endpoint are we integrating with and what is the expected response body?
- [ ] Do we want to capture analytics for this projects (i.e. update GA, add logs to grafana/datadog etc)?If yes, what metrics would we like to capture?

</details>

<details>
  <summary>Toggle QA Checklist</summary> (Not Applicable)

- [ ] Create test plan
  -  **Template: QA Test Plan**
- [ ] Identify test data needs
- [ ] Gotten & incorporated test plan review
- [ ] QA Complete
- [ ] Waygate tasks complete
- [ ] Added new feature cases to the RC in TestRail
- [ ] Added feature cases to appropriate section of active cases in TestRail
- [ ] Spun up a (future) UI automation ticket for new feature
</details>

**Comment by StacyB2023** [comment]
@bischoffa This is a case where the product is shipped but we are still setting up analytics. So yes, technically it is complete as on the Kanban board but this sprint 80 is being used to set up analytics. Once that is complete, I will close this epic.

cc: @g-opzt

**Comment by StacyB2023** [comment]
Deep Linking Dashboard: https://lookerstudio.google.com/u/0/reporting/ee7b19ce-7c3a-4a0c-a5fe-1a5dcbdffc58/page/p_tviavj4x8c

**Comment by StacyB2023** [comment]
Closing this MVP epic
